### PR TITLE
Several Inference Endpoint fixes

### PR DESCRIPTION
**Pull request by tengomucho**
# What does this PR do?

- Removed variables form entrypoint.sh, they are passed to the launcher via env vars
- Prevent issues with numpy 2.0 version
- correct TGI version
- add GKE ulimit command
- correct CachedBatch serialization when it's None (it was causing health call to crash)
- prefill and decode input pre-processing done in CPU to prevent wasting memory and time in compilation on TPU
- warmup clears after prefill
- fix clear implementation

### CODE REVIEW

File: text-generation-inference/server/text_generation_server/generator.py [hunk 1]
```diff
@@ -337,45 +355,59 @@ def prefill(self, batch: Batch) -> Tuple[List[Generation], CachedBatch]:
...
+            tokenized_input.input_ids = tokenized_input.input_ids[:, -slot.truncate:][:self.model.config.sequence_length]
+            tokenized_input.attention_mask = tokenized_input.attention_mask[:, -slot.truncate:][:self.model.config.sequence_length]
+            tokenized_inputs.append(tokenized_input)
+            # Update the maximum sequence length
+            seq_length = max(seq_length, tokenized_input.input_ids.size(-1))
+
+        batch_size = len(self.slots)
+        # Initialize input_ids and attention_mask with padding (to make them all the same size)
+        input_ids = torch.full((batch_size, seq_length), self.tokenizer.pad_token_id, dtype=torch.int64)
+        attention_mask = torch.full((batch_size, seq_length), 0, dtype=torch.int64)
```
Comment (mfuntowicz) [inline text-generation-inference/server/text_generation_server/generator.py:383]: qq: Can't we make it use `int32` for `input_ids` and `attention_mask` ?

File: text-generation-inference/server/text_generation_server/generator.py [hunk 2]
```diff
@@ -565,27 +604,25 @@ def filter(self, batch_id: int, request_ids: List[int]) -> CachedBatch:
...
+        if batch_id is not None:
+            keep_ids = [slot.id for slot in self.slots if slot.batch_id != batch_id]
+        return self._clear(keep_ids)
+
+    def _clear(self, keep_slot_ids: List):
         for slot in self.slots:
-            if slot.state != Slot.State.EMPTY and slot.request_id not in request_ids:
-                logger.debug(f"Removing request {slot.request_id}")
+            if slot.state != Slot.State.EMPTY and slot.id not in keep_slot_ids:
+                logger.info(f"Removing slot {slot.id} with request {slot.request_id}")
```
Comment (mfuntowicz) [inline text-generation-inference/server/text_generation_server/generator.py:621]: `debug` might be enough for this imo

**Review by mfuntowicz — APPROVED** [review]

File: text-generation-inference/server/text_generation_server/generator.py (hunk 1 above)
Comment (tengomucho) [reply text-generation-inference/server/text_generation_server/generator.py:383]: Umh, maybe we could, but the price would be a frequent cast between int32 and int64, because PB integers are serialized in int64. I think it's not worth doing it here, we could look at the advantages/disadvantages in another PR.
